pypeit/reduce.py

```diff
@@ -528,35 +535,73 @@ def global_skysub(self, scaleImg=None, skymask=None, update_crmask=True, trim_ed
         # Mask objects using the skymask? If skymask has been set by objfinding, and masking is requested, then do so
         skymask_now = skymask if (skymask is not None) else np.ones_like(self.sciImg.image, dtype=bool)
 
-        # Select the edges to use: Selects the edges tweaked by the
-        # illumination profile if they're present; otherwise, it
-        # selects the original edges from EdgeTraceSet. To always
-        # select the latter, use the method with `original=True`.
-        # TODO -- JXP removed these from KCWI4..
-        #original = True if ((trim_edg[0] == 0) and (trim_edg[1] == 0)) else False
-        #left, right = self.slits.select_edges(original=original)
-        #left, right = self.slits.select_edges()
-
         if self.par['reduce']['skysub']['joint_fit']:
             msgs.info("Performing joint global sky subtraction")
             thismask = (self.slitmask != 0)
-            inmask = (self.sciImg.fullmask == 0) & thismask & skymask_now
-            wavenorm = self.waveimg / np.max(self.waveimg)
+            inmask = ((self.sciImg.fullmask == 0) & thismask & skymask_now).astype(np.bool)
+            # Convert the wavelength image to A/pixel, registered at pixel 0 (this gives something like
+            # the tilts frame, but conserves wavelength position in each slit)
+            tilt_wave = (self.waveimg - self.waveimg.min()) / (self.waveimg.max() - self.waveimg.min())
             # Find sky
-            # TODO :: JXP removed the left and right (non trimmed) edges (see above). This might not allow the whole slit to be used
-            scalefact = scaleImg + (scaleImg == 0)
             self.global_sky[thismask] \
-                = skysub.global_skysub(self.sciImg.image/scalefact, self.sciImg.ivar, wavenorm,
+                = skysub.global_skysub(self.sciImg.image, self.sciImg.ivar, tilt_wave,
                                        thismask, self.slits_left, self.slits_right, inmask=inmask,
                                        sigrej=sigrej, trim_edg=trim_edg,
                                        bsp=self.par['reduce']['skysub']['bspline_spacing'],
                                        no_poly=self.par['reduce']['skysub']['no_poly'],
-                                       pos_mask=(not self.ir_redux), show_fit=True)#show_fit)
-            # Apply the scaling factor to the sky image
-            self.global_sky *= scaleImg
+                                       pos_mask=(not self.ir_redux), use_wave=True, show_fit=show_fit)
```

Comment:
use_wave is no longer an option to global_skysub, so technically this should crash? Have you tested this recently?